packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -60,23 +71,24 @@ class CupertinoPageTransition extends AnimatedWidget {
       );
 
   // When this page is coming in to cover another page.
-  final Animation<FractionalOffset> incomingPositionAnimation;
+  final Animation<FractionalOffset> _incomingPositionAnimation;
   // When this page is becoming covered by another page.
-  final Animation<FractionalOffset> outgoingPositionAnimation;
+  final Animation<FractionalOffset> _outgoingPositionAnimation;
+  final Animation<int> _incomingElevationAnimation;
   final Widget child;
 
   @override
   Widget build(BuildContext context) {
     // TODO(ianh): tell the transform to be un-transformed for hit testing
     // but not while being controlled by a gesture.
     return new SlideTransition(
-      position: outgoingPositionAnimation,
+      position: _outgoingPositionAnimation,
       child: new SlideTransition(
-        position: incomingPositionAnimation,
+        position: _incomingPositionAnimation,
         child: new PhysicalModel(
           shape: BoxShape.rectangle,
           color: _kBackgroundColor,
-          elevation: 32,
+          elevation: _incomingElevationAnimation.value,
```

Comment:
This is moot, but, if you were to have code like this, I would suggest adding a comment explaining the design and then adding a test to make sure it never regresses (e.g. if the parent animation is removed from the design).